include/unifex/scheduler_concepts.hpp

```diff
@@ -337,6 +337,21 @@ namespace _now {
 } // namespace _now
 using _now::now;
 
+namespace _current {
+  inline constexpr struct _scheduler {
+    auto schedule() const noexcept {
```

Comment:
I was thinking that this should forward `tag_invoke()` rather than have member `schedule()`. This way it could support time and priority schedulers as well.

Reply:
The meaning of `unifex::schedule()` is fixed. There is no dispatching with `tag_invoke` at this point. Supporting time schedulers is a good idea, though. I think a `schedule_after` member is the right way to support that, though.

It doesn't look like there is support yet for `unifex::schedule_at(TimePoint)` to be used with `current_scheduler`. I could add that.

I'm also tempted to move this functionality out of the customization points and into `current_scheduler` directly. The meaning of `unifex::schedule()` (with no arguments) is pretty unintuitive. But I suppose I can leave that for another time.

Reply:
I agree about `schedule()` - I am not sure about the `tag_invoke()` statement. I was thinking of a `tag_invoke()` overload that returns a sender that stores the cpo and arguments and uses them to call `tag_invoke()` in `connect()` with the target being `get_scheduler(receiver)`. I guess the challenge would be to constrain the `current_scheduler` `tag_invoke()` so that only schedule cpos were accepted. It might require some oddity like a common base tag for all the schedule cpos.

Reply:
Yes, something like that would be possible. I've been meaning to extend unifex's `tag_invoke` with CPO categories like I first envisioned. We can use that, e.g., instead of the `is_receiver_cpo_v` trait also.

But that's work for another day. I'm going to merge this as-is for now.

Reply:
I was wondering whether new scheduler-based algorithms should explicitly add support for the `current_scheduler` scheduler when they are added, rather than adding them as members of the scheduler here.

Adding it for the scheduler methods defined in here is probably ok, but when extending, say, the I/O or file scheduler operations, they should probably add customisations for `current_scheduler()` within those CPOs rather than keep extending this class (which would create all sorts of circular dependencies).